crates/ty_project/src/lib.rs

```diff
@@ -372,13 +392,28 @@ impl Project {
         }
     }
 
+    /// Returns `true` if the file should be checked.
+    ///
+    /// This depends on the project's check mode:
+    /// * For [`OpenFiles`], it checks if the file is open in the project
+    /// * For [`AllFiles`], it always returns `true`
+    ///
+    /// [`OpenFiles`]: CheckMode::OpenFiles
+    /// [`AllFiles`]: CheckMode::AllFiles
+    pub fn should_check_file(self, db: &dyn Db, file: File) -> bool {
+        match self.check_mode(db) {
+            CheckMode::OpenFiles => self.is_file_open(db, file),
+            CheckMode::AllFiles => true,
```

Comment:
I don't think using `true` here is correct because it would also return `true` for non project files (e.g. third party files!). We still need to test if the file is in `self.files(db)`